### PR TITLE
Add basic testing for the abstract FilterPlugin class

### PR DESCRIPTION
**Pull request by arcturus2**
### Description of the Change

A basic test to cover the abstract `FilterPlugin` class.

### Applicable Issues

#413

### CODE REVIEW

**Review by antares1470 — COMMENTED** [review]
Just curious @arcturus2 as to whether you intend to write unit tests for `FilterTypeParameterValue` (the static class that sits in `FilterType`)?

**Comment by arcturus2** [comment]
> Just curious @arcturus2 as to whether you intend to write unit tests for `FilterTypeParameterValue` (the static class that sits in `FilterType`)?

Thank you for the feedback @antares1470. I've managed to add more unit tests so please review when you can.

**Comment by arcturus2** [comment]
Fixed some code smells and the code coverage is looking better. I might do a bit more but happy for this to be reviewed as is for merging.

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/FilterPluginNGTest.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of validateString method, of class FilterTypeParameterValue.
+     */
+    @Test
+    public void testValidateStringWhenInvalid() {
+        System.out.println("validateString");
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/FilterPluginNGTest.java:158]: Minor thing but are you able to potentially alter the string here? Just so we're able to differentiate this test from the previous one if we need to to go through logs

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/FilterPluginNGTest.java (hunk 1 above)
Comment (arcturus2) [reply CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/FilterPluginNGTest.java]: Sure thing @antares1470 I've just updated it and its ready for another review.

**Review by antares1470 — APPROVED** [review]

**Pull request merged**